cider/exec/nextgen/context/CodegenContext.h

```diff
@@ -71,21 +71,24 @@ class CodegenContext {
         : ctx_id(id), name(n), type(t) {}
   };
 
-  using BatchDescriptorPtr = std::shared_ptr<BatchDescriptor>;
+  void setJITModule(std::unique_ptr<jitlib::LLVMJITModule> jit_module) {
```

Comment:
We should always use base class jitlib::JITModule. BTW, if JITModule will be passed between multiple functions, better to use shared_ptr, same as ```JITFunctionPtr```.

Reply:
module is internal used only